app/delegate-server/Contract/Commit.purs

```diff
@@ -0,0 +1,69 @@
+module DelegateServer.Contract.Commit
+  ( commitCollateral
+  , commitStandingBid
+  ) where
+
+import Contract.Prelude
+
+import Contract.Monad (Contract)
+import Contract.Transaction
+  ( BalancedSignedTransaction(BalancedSignedTransaction)
+  , _vkeys
+  , _witnessSet
+  , signTransaction
+  , submit
+  )
+import Control.Monad.Reader (asks)
+import Data.Argonaut (encodeJson)
+import Data.Lens ((.~))
+import Data.Lens.Common (simple)
+import Data.Lens.Iso.Newtype (_Newtype)
+import DelegateServer.HydraNodeApi.Http (commit)
+import DelegateServer.HydraNodeApi.Types.Commit
+  ( CommitUtxoMap
+  , mkCollateralCommit
+  , mkStandingBidCommit
+  )
+import DelegateServer.Lib.Json (printJson)
+import DelegateServer.Lib.ServerConfig (mkLocalhostHttpServerConfig)
+import DelegateServer.Lib.Wallet (withWallet)
+import DelegateServer.State (AppM, readAppState, runContract)
+import HydraAuctionOffchain.Contract.QueryUtxo (queryStandingBidUtxo)
+import HydraAuctionOffchain.Contract.Validators (mkStandingBidValidator)
+
+commitStandingBid :: AppM Unit
+commitStandingBid = do
+  auctionInfo <- unwrap <$> readAppState _.auctionInfo
+  collateralUtxo <- readAppState _.collateralUtxo
+  standingBidValidator <- runContract $ mkStandingBidValidator auctionInfo.auctionId
+    auctionInfo.auctionTerms
+  bidUtxo <- runContract $ queryStandingBidUtxo auctionInfo
+  case bidUtxo of
+    Just (utxo /\ _) | Just bidCommit <- mkStandingBidCommit utxo standingBidValidator -> do
+      liftEffect $ log $ "Ready to commit standing bid: " <>
+        printJson (encodeJson bidCommit)
+      commitUtxos $ bidCommit <> mkCollateralCommit collateralUtxo
+    _ -> pure unit
```

Comment:
Should we error id bidUtxo was not found?

Reply:
Yes, this code is WIP and some work is planned to enhance its robustness.